statsforecast/core.py

```diff
@@ -678,10 +703,9 @@ def fit_predict(
         ----------
         h : int
             Forecast horizon.
-        df : pandas.DataFrame | polars.DataFrame, optional (default=None)
-            DataFrame with columns [`unique_id`, `ds`, `y`] and exogenous variables.
-            If None, the `StatsForecast` class should have been instantiated
-            using `df`.
+        df : pandas or polars DataFrame, optional (default=None)
+            DataFrame with columns [`id_col`, `time_col`, `target_col`] and exogenous.
+            If None, the `StatsForecast` class should have been instantiated using `df`.
         X_df : pandas.DataFrame | polars.DataFrame, optional (default=None)
             DataFrame with [`unique_id`, `ds`] columns and `df`'s future exogenous.
```

Comment:
Should we also update the `X_df` documentation?

Reply:
Added the templating in 678f3c1